src/api/osaamiset.ts

```diff
@@ -0,0 +1,86 @@
+import { client } from './client';
+import { components } from './schema';
+
+type OsaaminenDto = components['schemas']['OsaaminenDto'];
+const cache = new Map<string, OsaaminenDto>();
+const maxQueryLength = 4096;
+const queryParamOverhead = 5; // uri=...&
+
+export const osaamiset = {
+  /**
+   * Combines objects referencing osaamiset with osaamiset.
+   *
+   * @param objects Collection of objects to combine with osaamiset.
+   * @param key Property accessor for the osaaminen uri.
+   * @param combiner Function to combine the object with the osaaminen.
+   * @param signal Abort signal.
+   * @returns An array of objects combined with osaamiset.
+   */
+  combine: async <T, R>(
+    objects: Iterable<T> | undefined,
+    key: (o: T) => string,
+    combiner: (object: T, osaaminen: OsaaminenDto) => R,
+    signal?: AbortSignal,
+  ) => {
+    if (!objects) {
+      return [];
+    }
+    const chunks: string[][] = [];
+    const result: R[] = [];
+    const seen = new Set<string>();
+
+    let len = 0;
+    for (const obj of objects) {
+      const uri = key(obj);
+      if (cache.has(uri) || seen.has(uri)) {
+        continue;
+      }
+      seen.add(uri);
+      if (len == 0 || len > maxQueryLength) {
+        chunks.push([uri]);
+        len = uri.length + queryParamOverhead;
+      } else {
+        chunks[chunks.length - 1].push(uri);
+        len += uri.length + queryParamOverhead;
+      }
+    }
+
+    if (chunks.length > 0) {
+      (
+        await Promise.all(
+          chunks.map((chunk) =>
+            client.GET('/api/osaamiset', {
+              signal,
+              params: { query: { uri: chunk, sivu: 0, koko: chunk.length } },
+            }),
+          ),
+        )
+      )
+        .flatMap((result) => result.data?.sisalto ?? [])
+        .forEach((osaaminen) => cache.set(osaaminen.uri, osaaminen));
+    }
```

Comment:
Does the chunking and combining results warrant a unit test? Any interesting corner cases?